File: LLM-Translations/GPT-4/avatar/Java/Go/codeforces_518_B.go
package main

import (
	"fmt"
	"strings"
	"unicode"
)

func main() {
	var s, t string
	fmt.Scan(&s, &t)

	ura := 0
	opa := 0

	tmap := make(map[rune]int)
	for _, ch := range t {
		tmap[ch]++
	}

	srunes := []rune(s)
	length := len(srunes)
	for i := 0; i < length; i++ {
		ch := srunes[i]
		num, ok := tmap[ch]
		if !ok {
			continue
		}
		if num == 1 {
			delete(tmap, ch)
		} else {
			tmap[ch] = num - 1
		}
		srunes[i] = 'Я'
		ura++
	}

	for i := 0; i < length; i++ {
		ch := srunes[i]
		rch := reverse(ch)
		num, ok := tmap[rch]
		if !ok {
			continue
		}
		if num == 1 {
			delete(tmap, rch)
		} else {
			tmap[rch] = num - 1
		}
		opa++
	}

	fmt.Println(ura, opa)
}

func reverse(ch rune) rune {
	if unicode.IsLower(ch) {
		return unicode.ToUpper(ch)
	}
	return unicode.ToLower(ch)
}

// //End of Code.

File: LLM-Translations/GPT-4/avatar/Java/Python/codeforces_518_B.py
import sys
from collections import defaultdict

def main():
    s = input().strip()
    t = input().strip()
    ura = 0
    opa = 0
    tmap = defaultdict(int)
    for ch in t:
        tmap[ch] += 1
    s = list(s)
    length = len(s)
    for i in range(length):
        ch = s[i]
        num = tmap[ch]
        if num == 0:
            continue
        if num == 1:
            del tmap[ch]
        else:
            tmap[ch] -= 1
        s[i] = 'Я'
        ura += 1
    for i in range(length):
        ch = s[i]
        rch = reverse(ch)
        num = tmap[rch]
        if num == 0:
            continue
        if num == 1:
            del tmap[rch]
        else:
            tmap[rch] -= 1
        opa += 1
    print(ura, opa)

def reverse(ch):
    if ch.islower():
        return ch.upper()
    else:
        return ch.lower()

if __name__ == "__main__":
    main()


